ir_beacon/plugins/IRSensorPlugin.cc

```diff
@@ -223,10 +241,63 @@ void IRMaterialHandler::TagIRVisual(rendering::VisualPtr _vis)
   obj->getUserObjectBindings().setUserAny(Ogre::Any(std::string("ir_glow")));
 }
 
+/////////////////////////////////////////////////
+void IRMaterialHandler::RemoveTag(rendering::VisualPtr _vis)
+{
+  if (!_vis)
+    return;
+
+  // tag child visuals too
+  for (unsigned int i = 0; i <_vis->GetChildCount(); ++i)
+  {
+    rendering::VisualPtr child = _vis->GetChild(i);
+    this->RemoveTag(child);
+  }
+
+  Ogre::SceneNode *node = _vis->GetSceneNode();
+  if (!node)
+    return;
+
+  if (node->numAttachedObjects() == 0)
+    return;
+
+  Ogre::MovableObject *obj = node->getAttachedObject(0);
+  if (!obj)
+    return;
+
+  Ogre::Any objUserAny = obj->getUserObjectBindings().getUserAny();
+  if (objUserAny.isEmpty())
+    return;
+
+  std::string userAny = "";
+  try
+  {
+    userAny = Ogre::any_cast<std::string>(objUserAny);
+  }
+  catch(Ogre::Exception &e)
+  {
+    gzerr << "Unable to cast Ogre user data" << std::endl;
+    return;
+  }
+
+  // targets will have the user data "ir_glow"
+  if (userAny == "ir_glow")
+  {
+    _vis->GetScene()->RemoveVisual(_vis);
+  }
+}
+
 
 /////////////////////////////////////////////////
 void IRMaterialHandler::PreRender()
 {
+  // remove existing targets
+  for (auto vis : this->targetVisuals)
+  {
+    this->RemoveTag(vis);
```

Comment:
Is this going to remove the tag at every render iteration to then be added back down below?

Reply:
good catch, I added a flag to make sure it only removes the visuals only when new targets are set. af4ae88